README.rst

```diff
@@ -78,6 +78,31 @@ The ``asv preview`` command will give the URL of the local web server (e.g.
 http://127.0.0.1:21331) - go to this address in your favorite browser to see
 the results.
 
+Running benchmarks with specific versions of dependencies
+---------------------------------------------------------
+
+If you would like to run the benchmarks against specific version(s) of an
+installed dependency (or dependencies), modify ``asv.conf.json`` to add 
+``matrix.req`` with the desired version(s):
+
+.. code-block:: json
+
+    {
+      ...,
+      "matrix": {
+        "req": {
+          "matplotlib": [
+            "3.5.1"
+          ],
+          "numpy": [
+            "1.26.0",
+            "2.0.0rc1"
+          ]
+        }
+      },
+      ...
+    }
+
```

Comment:
This works in Markdown but not sure about RST. 🤷‍♀️ 

```diff
--- a/asv.conf.json
+++ b/asv.conf.json
@@ -6,6 +6,17 @@
   "install_command": [
     "pip install . matplotlib scipy"
   ],
+  "matrix": {
+    "req": {
+      "matplotlib": [
+        "3.5.1"
+      ],
+    "numpy": [
+      "1.26.0",
+      "2.0.0rc1"
+      ]
+    }
+  },
   "branches": ["main"],
   "show_commit_url": "http://github.com/astropy/astropy/commit/",
   "pythons": ["3.11"],
```

Reply:
I think GitHub's rendering of RST is a bit wonky (maybe they want us to use markdown instead). We could change the filetype to markdown, possibly

Reply:
Would you mind doing that here? It is technically out of scope so you can say no. I just noticed that the Python formatting is also off. 🤷‍♀️

Reply:
sure thing! Here: https://github.com/astropy/astropy-benchmarks/blob/011808fbf52215ac7b13517246a013bdceeb7fc8/README.md